### DISCUSSION TITLE
Invalid refresh token?

### DISCUSSION THREAD

**Discussion by TUBEOF** [spotify-web-api-java/spotify-web-api-java#398] [Q&A]
When requesting a refresh and access token from Spotify using OAuth, the refresh and access tokens are invalid somehow?
Even when trying to refresh the token using the refresh token I've got 1 second ago, I get an "Invalid refresh token" error.
I compared the used tokens, but there are 100% the same as I got from Spotify:
![image](https://github.com/spotify-web-api-java/spotify-web-api-java/assets/84939906/cbff3df3-d42f-48d5-8a10-ae4d80de41ee)

Refresh Request looks like this:
```
        AuthorizationCodeRefreshRequest authorizationCodeRefreshRequest = spotifyApi.authorizationCodeRefresh().build();
        System.out.println("ACCESS: " + spotifyApi.getAccessToken());
        System.out.println("REFRESH: " + spotifyApi.getRefreshToken());
        AuthorizationCodeCredentials authorizationCodeCredentials = authorizationCodeRefreshRequest.execute();
```

Everything works fine in my other projects, but in this case somehow not and everything is the same?

**Comment by dargmuesli** [comment]
I think the four lines provided are not sufficient to understand the issue.

**Reply by TUBEOF** [reply]
Getting refresh and access token using OAuth Code:
```java
AuthorizationCodeRequest authorizationCodeRequest = spotifyApi.authorizationCode(httpServerExchange.getQueryParameters().get("code").getFirst()).build();
AuthorizationCodeCredentials authorizationCodeCredentials = authorizationCodeRequest.execute();
```

Then, when i try to request some data, using the access token, i get "Invalid access token":
```java
            GetTheUsersQueueRequest getTheUsersQueueRequest;
            getTheUsersQueueRequest = spotifyApi.getTheUsersQueue().build();
            PlaybackQueue playbackQueue = getTheUsersQueueRequest.execute();
```

So trying to refresh the access token using the refresh token, but getting "Invalid refresh token"
```
AuthorizationCodeRefreshRequest authorizationCodeRefreshRequest = spotifyApi.authorizationCodeRefresh().build();
AuthorizationCodeCredentials authorizationCodeCredentials = authorizationCodeRefreshRequest.execute();
```

How? I got both tokens like a few seconds.

**Comment by TUBEOF** [comment]
I also added the user as a tester for the Application at the Spotify Developer Portal, so this cannot be the issue.
